torch/nn/functional.py

```diff
@@ -3737,6 +3737,42 @@ def triplet_margin_loss(anchor, positive, negative, margin=1.0, p=2, eps=1e-6, s
                                      swap, reduction_enum)
 
 
+def triplet_margin_loss_with_distance(anchor, positive, negative, distance_function=None, is_similarity_function=False,
+                                      margin=1.0, swap=False, reduction="mean"):
+    # type: (Tensor, Tensor, Tensor, Optional[Callable[[Tensor, Tensor], Tensor]], bool, float, bool, str) -> Tensor
```

Comment:
Can this use Python3-style typing?

Reply:
Curious what the standard is for this file -- I saw a lot of py2 typing on the other functions.

Reply:
If the rest of the file is Python2-style typing then it's fine.